src/compiler/checker.ts

```diff
@@ -20964,6 +20985,25 @@ namespace ts {
                         flags |= ModifierFlags.Async;
                         lastAsync = modifier;
                         break;
+
+                case SyntaxKind.OverrideKeyword:

```

Comment:
Done for (1), (2).  As for (3), I still don't really understand what consecutive declaration checks mean.  Please propose a test class that illustrates it if possible.